HW0/P5/P5.txt

```diff
@@ -1 +1,18 @@
-# Your answers here.
+# Student: Xingchi Dai
+
+Q1: (255 secs) For 256 bags, one calculator, it will take one person 255 seconds to count all pennies. 
+
+Q2: (34 secs) We have 8 people. Everyone takes 32 nags and in 31 seconds, all bags will be counted. Then it will take three seconds to add them up.
+
+Q3: (8 secs). If we have infinite people, let assume everyone takes two bags, then everyone takes two numbers to add up. It will take 8 seconds to finish counting. 
+
+Q4: For multiple persons, the trends would equals to log2(bags_numbers) while the single person equals to n-1
+
+Q5: (15 Seconds), in this case, if two people talk, we will add 1 sec. Therefore, 7 seconds will be added. 
+
+Q6: (272 s, 270s) If we gave bags to 256 people, it will take 256 s for handling, 8 secs to communicate each other and calculating. The total will be 272 secs. The same thing for two bags each person, it will take 270 s.
```

Comment:
I agree with your logic (handing, calculating), but I think you're not considering the fact that the communication can happen at the same time as we hand out the bags (ie, give the first four bags out, then after two seconds, those four people have summed to a value). Does that make sense?